src/VisualStudio/Core/Def/TaskList/ProjectExternalErrorReporter.cs

```diff
@@ -282,15 +261,26 @@ public void ReportError2(
         int iEndColumn,
         string bstrFileName)
     {
+
+        // make sure we have error id, otherwise, we simple don't support
+        // this error
+        if (bstrErrorId == null)
+        {
+            // record NFW to see who violates contract.
+            FatalError.ReportAndCatch(new Exception("errorId is null"));
+            return;
+        }
+
         _taskQueue.AddWork(ReportErrorAsync);
 
         async Task ReportErrorAsync(CancellationToken cancellationToken)
         {
-            // first we check whether given error is something we can take care.
-            if (!await CanHandleAsync(bstrErrorId, cancellationToken).ConfigureAwait(false))
+            // we accept all compiler diagnostics
+            if (!bstrErrorId.StartsWith(_errorCodePrefix) &&
+                !await DiagnosticProvider.IsSupportedDiagnosticIdAsync(
+                    _projectId, bstrErrorId, cancellationToken).ConfigureAwait(false))
```

Comment:
What would it look like to have some code that updates the list of IDs as an in-proc cache which we can still check synchronously?